keps/sig-auth/4872-harden-kubelet-cert-validation/kep.yaml

```diff
@@ -0,0 +1,40 @@
+title: Harden Kubelet Serving Certificate Validation in Kube-API server
+kep-number: 4872
+authors:
+  - "@g-gaston"
+owning-sig: sig-auth
+participating-sigs:
+status: provisional
+creation-date: 2024-09-24
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+see-also:
+replaces:
+
+# The target maturity stage in the current dev cycle for this KEP.
+stage: alpha|beta|stable
+
+# The most recent milestone for which work toward delivery of this KEP has been
+# done. This can be the current (upcoming) milestone, if it is being actively
+# worked on.
+latest-milestone: ""
+
+# The milestone at which this feature was, or is targeted to be, at each stage.
+milestone:
+  alpha: ""
+  beta: ""
+  stable: ""
+
+# The following PRR answers are required at alpha release
+# List the feature gate name and the components for which it must be enabled
+feature-gates:
+  - name: KubeletCertCNValidation
```

Comment:
How about: `NodeCertificateNameValidation`?

The API server is verifying a certificate presented by a node, whatever software that node is running (which, OK, is very likely to be `kubelet`).